x/exchange/module/module.go

```diff
@@ -0,0 +1,156 @@
+package module
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	sdkclient "github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+
+	"github.com/provenance-io/provenance/x/exchange"
+	"github.com/provenance-io/provenance/x/exchange/client/cli"
+	"github.com/provenance-io/provenance/x/exchange/keeper"
+	"github.com/provenance-io/provenance/x/exchange/simulation"
+)
+
+var (
+	_ module.AppModule           = AppModule{}
+	_ module.AppModuleBasic      = AppModuleBasic{}
+	_ module.AppModuleSimulation = AppModule{}
+)
+
+type AppModule struct {
+	AppModuleBasic
+	keeper keeper.Keeper
+}
+
+func NewAppModule(cdc codec.Codec, exchangeKeeper keeper.Keeper) AppModule {
```

Comment:
This should also be with the AppModule methods.

Reply:
Moved this too.